custom_components/tuya_local/devices/faroro_pet_feeder.yaml

```diff
@@ -0,0 +1,135 @@
+name: Wi-Fi Pet Feeder
+products:
+  - id: vglnuveujzyrpku9
+    name: Faroro Automatic Cat Feeder
+primary_entity:
+  entity: number
+  name: Manual Feed
+  icon: "mdi:food-drumstick"
+  mode: "box"
+  dps:
+    - id: 3
+      name: value
+      type: integer
+      persist: false
+      range:
+        min: 1
+        max: 12
+secondary_entities:
+  - entity: sensor
+    name: Feed Status
+    icon: "mdi:paw"
+    class: enum
+    dps:
+      - id: 4
+        name: sensor
+        type: string
+        mapping:
+          - dps_val: standby
+            value: Standby
+          - dps_val: feeding
+            value: Feeding
+  - entity: button
+    name: Factory Reset
+    icon: "mdi:lock-reset"
+    category: diagnostic
+    dps:
+      - id: 9
+        type: boolean
+        name: button
+  - entity: sensor
+    name: Last Feed Amount
+    icon: "mdi:food-drumstick"
+    category: diagnostic
+    dps:
+      - id: 15
+        name: sensor
+        type: integer
+  - entity: sensor
+    name: Voice Times
+    icon: "mdi:account-voice"
+    category: config
+    dps:
+      - id: 18
+        name: sensor
+        type: integer
+  - entity: switch
+    name: Indicator Light
+    icon: "mdi:lightbulb"
+    category: config
+    dps:
+      - id: 19
+        name: switch
+        type: boolean
+        mapping:
+          - dps_val: false
+            icon: "mdi:lightbulb-outline"
+  - entity: sensor
+    name: Battery Level
+    icon: "mdi:battery"
+    category: diagnostic
```

Comment:
This could be made a `class: battery` sensor by changing the mappings to something like 80, 50, 20 (maybe lower for the value "no", I usually avoid 100 and 0 because they can be misleading when the steps are this coarse but "no" does imply that it really means 0).

Then the name and icon can be left to Home Assistant to manage.